src/vswitch.c

```diff
@@ -64,6 +68,15 @@ SEXP vswitchR(SEXP x, SEXP values, SEXP outputs, SEXP na, SEXP nthreads) {
   int nprotect=0;
   SEXP ans = R_NilValue;
   SEXPTYPE type_o = UTYPEOF(outputs);
+  SEXP xans = R_NilValue, vans = R_NilValue;
+  bool utfcon = false;
+  if (pchkenc && type_x == STRSXP) {
+    if (isMixEnc(x) || isMixEnc(values) || getCharCE(STRING_PTR(x)[0]) != getCharCE(STRING_PTR(values)[0])) {
```

Comment:
Here you assume the vector is at least length one. Any chance that a zero-length vector being passed into the routine can lead to segfaults?

Other parts look good to me. Thanks.

Reply:
For the zero length vector, I tried to make the function crash but did not manage to.

Reply:
If zero-length vectors can reached to these lines, I suggest to handle it carefully. [An example](https://github.com/Rdatatable/data.table/issues/4060#issuecomment-556011718)  is 

```r
Rcpp::cppFunction("int getlength(SEXP x) { return LENGTH(VECTOR_ELT(x,0)); }")
getlength(list())
```

where garbage values generated on R3.4 but always zero on R3.5 - 3.6.

Reply:
I added a check on the length of x and values that should prevent a zero-length vector to reach these lines.